README
YAAI is an Asterisk Connector for your SugarCRM Instance.

Documentation for this project is on the Wiki pages.

If you have questions, please post them in the issues section.


Demo Video
==========
http://www.youtube.com/watch?v=UAdxUNlPDzQ

Links to Manuals, Project TODO, Etc
==========
https://github.com/blak3r/yaai/wiki/_pages

YAAI Discussion Forum for asking questions, reporting bugs, feature requests, etc.
==========
https://github.com/blak3r/yaai/issues

Donate Link
==========
http://www.blakerobertson.com/yaai-donation-page/

Branches
=========
Currently, the current stable release is maintained in the master branch.  
Dev versions are maintained in other branches such as "V3".

Mailing List
=============
Subscribe to the mailing list here for news: http://eepurl.com/rmdML



RELEASE-NOTES
IMPORTANT: when upgrading, please see the instructions in the User Manual <http://www.github.com/blak3r/yaai/wiki/User-Manual>

##Version 3.6.0 (2013-07-28)
* Enhancements
** Click To Dial is more intelligent, it always relates call to the record clicked. Before if multiple matching phone numbers, before if phone number matched multiple contacts it showed radio buttons.
** New config option "Max Call Popups" controls the maximum number of popups that will appear on a users screen.
** The 'X' button is easier to click to close call popups.
** (Basic/Pro Only) - Calls to Queues and RingGroups now popup on callee's screen before answering.
* Bug Fixes
** dial_events_log in admin config now works.
** Admin Checkboxes always appeared as checked.


##Version 3.5.4 (2013-07-08)
* New User Interface
* Ability to specify which contact and account phone fields are searched.
* Major Performance Improvements
* Improvements to support for ring groups and queues.
* Ability to relate calls to a contact or account.
* Account Matching in Popups.
* Improvements to Admin Pages to have checkboxes instead of textboxes for boolean config elements.
* Cleaner formatting of asterisk log statements.
* New dial events log


##Version 2.6.0 (2012-11-03)##

### NEW Features ###
User can have multiple extensions (comma separated) Issue #22
Calls can now be related to contacts created even after the call is hung up.
Short Call Status allow the user to specify whether or not missed calls get their Call Status set to "Missed" or "Held".  If it's Missed, then the call will show up as an open task.  I set the default ot be Held for future installs.

### Improvements / Bug Fixes ###
Various bug fixes related to AJAX-UI
Improved Daemon Script
Added lots more config options.
Ability to disable OpenCNAM integration.
Fix IE bug where multiple click to dial icons would appear in contact list view.
Fix for Issue #75, Outbound Caller Id not set
Improved compatibility for POTS asterisk (ie not voip).  Outbound calls now pull outbound phone number in properly from dialstring.
IE8 compatibility for Click to Dial
Click-2-Dial now works when Letrium Quick Edit Detail View is installed.
Fix for Issue #25, Calls are related to contacts created after call ends.
Fix to work with ZenoSMS
Better AMI Connection Diagnostics / Markdown Logging
Retries soap login indefinitely.
Updated time handling to fix #40.
Fixed a minor localization bug.
Adds support for LDAP auth (Issue #59)
Fixed bug in Stripping Inbound / Outbound Prefixes.
Deletes calls_cstm now also.
Fixed open_cnam print error
Adds Support For nested RingGroups, still has dev debugs on
Merge pull request #20 from superles/patch-1
Merge pull request #49 from trustmaster/master
Fixed #44: using raw English call statuses as before localization.
Merge branch 'master' of git://github.com/blak3r/yaai
Merge pull request #42 from kerio-fstedronsky/patch-1
Fix blak3r/yaai#41
Cleaned up some PHP Notices
Changed default search length to 10 digits.
Fix for Issue #69 and Issue #66
Fix for #40 timestamp handling
LDAP Auth Supported
Fixed Bug where Transfer didn't use context specified in the config.



## Version 2.4 (2012-07-22)
Fixes a bug where AJAX Calls prevented popups from closing.
Adds support for sugar 6.5, fixes some bugs, adds Russian localization.



## Version 2.3 (2012-05-09)

### New Features
* Added Transfer Call Support
* Improved UI on call popups to allow for selecting which contact the call
should be related to when multiple contacts match.
* Added OpenCNAM.com support for Caller ID
* Ability to manually relate a call to particular contact with a popup
* Relates to Account instead of a contact if number matches an account
number or if there are multiple contacts with the same phone number from
the same account.
* Doesn't log account to any contact if multiple contacts have the same
phone number and they belong to two different accounts.

### Changes:
* Lots of minor bug fixes.
* Module Loader automatically updates database
* Improved stability of asteriskLogger
* Performance Improvements.



## Version 2.0.3 (2012-04-03)

* Brand New \\\"gchat\\\" inspired UI.
* Added more configurable options to support custom dial plans.
* Supports multiple calls, ability to leave notes for up to an hour after
call has hungup.
* Improved support for Ring Groups which call an external phone number
(like cell phones).
* Improved stability of asterisk logger, doesn\\\'t hang as much
* Lots of misc bug fixes.

misc/en_us.lang.php
<?php
$GLOBALS['app_list_strings']['duration_intervals']=array (
  '' => '',
  0 => '00',
  1 => '01',
  2 => '02',
  3 => '03',
  4 => '04',
  5 => '05',
  6 => '06',
  7 => '07',
  8 => '08',
  9 => '09',
  10 => '10',
  11 => '11',
  12 => '12',
  13 => '13',
  14 => '14',
  15 => '15',
  16 => '16',
  17 => '17',
  18 => '18',
  19 => '19',
  20 => '20',
  21 => '21',
  22 => '22',
  23 => '23',
  24 => '24',
  25 => '25',
  26 => '26',
  27 => '27',
  28 => '28',
  29 => '29',
  30 => '30',
  31 => '31',
  32 => '32',
  33 => '33',
  34 => '34',
  35 => '35',
  36 => '36',
  37 => '37',
  38 => '38',
  39 => '39',
  40 => '40',
  41 => '41',
  42 => '42',
  43 => '43',
  44 => '44',
  45 => '45',
  46 => '46',
  47 => '47',
  48 => '48',
  49 => '49',
  50 => '50',
  51 => '51',
  52 => '52',
  53 => '53',
  54 => '54',
  55 => '55',
  56 => '56',
  57 => '57',
  58 => '58',
  59 => '59',
);